src/constructs/sns/sns-topic.ts

```diff
@@ -1,15 +1,13 @@
 import { Topic } from "@aws-cdk/aws-sns";
-import type { CfnTopic, TopicProps } from "@aws-cdk/aws-sns";
+import type { TopicProps } from "@aws-cdk/aws-sns";
 import type { GuStack } from "../core";
+import { GuMigratingResource } from "../core/migrating";
 
-interface GuSnsTopicProps extends TopicProps {
-  overrideId?: boolean;
-}
+interface GuSnsTopicProps extends TopicProps, GuMigratingResource {}
 
 export class GuSnsTopic extends Topic {
```

Comment:
We might want to implement `GuStatefulConstruct` here too? Stack A might use an SNS topic as way to accept input from stack B, so replacing may be unsafe?